### PR TITLE
List learners paid, have exam, not enrolled

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?
Fix https://github.com/mitodl/micromasters/issues/5235

#### What's this PR do?
A management command to receive a file with a List of learners paid, have exam, not enrolled.

#### How should this be manually tested?
Run` ./manage.py list_paid_learners.py`

### CODE REVIEW

File: exams/management/commands/list_paid_learners.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            enrolled_user_ids = set(CachedEnrollment.get_cached_users(next_course_run))
+            paid_not_enrolled_users_ids = list(Line.objects.filter(
+                order__status__in=Order.FULFILLED_STATUSES,
+                course_key__in=course.courserun_set.values('edx_course_key'),
+            ).exclude(order__user__in=enrolled_user_ids).values_list('order__user', flat=True))
+            paid_not_enrolled_users = User.objects.filter(id__in=paid_not_enrolled_users_ids)
+            for user in paid_not_enrolled_users:
+                mmtrack = get_mmtrack(user, course.program)
+                if not has_to_pay_for_exam(mmtrack, course):
+                    username = get_social_username(user, BACKEND_MITX_ONLINE)
```
Comment (arslanashraf7) [inline exams/management/commands/list_paid_learners.py:53]: If `username` is None, it still adds the users to the list with None username which I think it shouldn't or might be overthinking in this.

**Review by arslanashraf7 — APPROVED** [review]
Looks good 👍  I added a small comment.

**Comment by arslanashraf7** [comment]
I just noticed, The original ticket has been closed https://github.com/mitodl/micromasters/issues/5235

File: exams/management/commands/list_paid_learners.py (hunk 1 above)
Comment (annagav) [reply exams/management/commands/list_paid_learners.py:53]: we do want to keep the record even if it is none, I think it is helpful for looking people up, even if the don't have mitxonline social auth.

**Comment by annagav** [comment]
@pdpinch In this PR we are combining all learners that need a coupon into one list. Instead of the "authorized" list, that included only people who have pass the course before, this will be a list of people who paid, did not use their attempt, and are not currently enrolled.

**Pull request merged**